### ISSUE TITLE
Installer fails to request sudo password while installing skills at /opt/mycroft/skills

### ISSUE THREAD

**Issue by AMDphreak** [MycroftAI/mycroft-core#1976]
Installing on Raspbian using the Linux instructions:
https://github.com/MycroftAI/mycroft-core/blob/dev/README.md

```$ ./dev_setup.sh```
```
The standard location for Mycroft skills is under /opt/mycroft/skills.
This script will create that folder for you.  This requires sudo
permission and might ask you for a password...
mkdir: cannot create directory ‘/opt/mycroft’: Permission denied
```
```$ sudo ./dev_setup.sh```
```
This script should not be run as root or with sudo.
If you really need to for this, rerun with --allow-root
```
If the warning is coming from the shell script, then why forbid the user from running as root, since you obviously acknowledge a potential use case for sudoing the entire install?

On the other hand, if the `sudo` utility is issuing this warning, then I definitely won't run this as root. So, how can I get the permissions prompt for sudo for the part that needs it?

**Comment by krisgesling** [comment]
Hi AMDphreak, 

First wanted to make sure you knew about our [Picroft image](https://mycroft.ai/to/picroft-image) that is based off Raspbian but has a lot of tweaks to make Mycroft work on a Raspberry Pi. If you're just trying to get setup on RPi then this is the strongly recommended approach.

For general linux installs, I ran into this sudo problem last night installing on a new Ubuntu box too so thanks for flagging it as an issue. 
Creating the folder is the only thing requiring sudo so if you manually create `/opt/mycroft/skills` and rerun  `./dev_setup.sh` without sudo you should be good to go. 

Hope you're up and running quickly, let us know how it goes via the [Forum](https://community.mycroft.ai) or [Chat](https://chat.mycroft.ai).

**Comment by AMDphreak** [comment]
> 
> 
> Hi AMDphreak,
> 
> First wanted to make sure you knew about our [Picroft image](https://mycroft.ai/to/picroft-image) that is based off Raspbian but has a lot of tweaks to make Mycroft work on a Raspberry Pi. If you're just trying to get setup on RPi then this is the strongly recommended approach.
> 
> For general linux installs, I ran into this sudo problem last night installing on a new Ubuntu box too so thanks for flagging it as an issue.
> Creating the folder is the only thing requiring sudo so if you manually create `/opt/mycroft/skills` and rerun `./dev_setup.sh` without sudo you should be good to go.
> 
> Hope you're up and running quickly, let us know how it goes via the [Forum](https://community.mycroft.ai) or [Chat](https://chat.mycroft.ai).

Hi Kris! I know about it. I tried it, and it did not detect my USB mic, and the Rpi was basically useless for programming and deploying anything else, so I decided to install Raspbian and then try to install Mycroft after.

**Comment by krisgesling** [comment]
Yeah the range of audio devices is probably our biggest challenge. We have an update coming that will make this much easier, but I know that doesn't help you right now...

If you still have the Picroft image around, this tweak to `mycroft.conf` has helped a lot of people:
https://community.mycroft.ai/t/fix-for-h600-logitech-usb-headset-and-picroft/5743/2

**Comment by AMDphreak** [comment]
I was under the impression most audio devices adhered to a standard, or at least a subset of their features would. Most USB mics work with Linux already, so wouldn't it be a simpler approach to use ALSA or jack for inputs? I'm not familiar with the details, but it seems this is reinventing the wheel and not doing a good job at it.

**Comment by qlixed** [comment]
Current master dev_setup.sh lack of the initialization of the SUDO script variable, that's why is failing, to resolve I modify the script adding that variable, here the current git diff output of that file:

```
$ git diff dev_setup.sh
diff --git a/dev_setup.sh b/dev_setup.sh
index d96843a7e0..122f111cbb 100755
--- a/dev_setup.sh
+++ b/dev_setup.sh
@@ -102,6 +102,10 @@ function get_YN() {
     done
 }
 
+if found_exe sudo ; then
+    SUDO=sudo
+fi
+
 if found_exe tput ; then
     GREEN="$(tput setaf 2)"
     BLUE="$(tput setaf 4)"
```
With just that it start to work. Or I think so because my install is just starting to go o ver the automated phase.

**Comment by forslund** [comment]
Indeed you are correct, it's only set when doing package checks further down in the code.

**Comment by AMDphreak** [comment]
@forslund it returned "sudo exists". Raspbian comes with sudo installed and configured to automatically validate without asking for a password. Guess you guys solved this.

**Comment by forslund** [comment]
I've merged the fix PR #1978. Big thanks to @qlixed for pointing this out. Closing the Issue
